cfgov/unprocessed/js/apps/financial-well-being/fwb-questions.js

```diff
@@ -1,83 +1,169 @@
-/* eslint no-global-assign: "off" */
 'use strict';
 
-var Analytics = require( '../../modules/Analytics' );
+const Analytics = require( '../../modules/Analytics' );
 
+/**
+ * Initialize the questionnaire
+ */
 function init() {
+  const questionElems = document.querySelectorAll( '#quiz-form fieldset' );
+  const radioButtons = document.querySelectorAll( '#quiz-form [type="radio"]' );
+  const submitButton = document.querySelector( '#submit-quiz' );
+  const questionStates = {};
+
+  /**
+   * Checks the selection status of a group of radio inputs,
+   * belonging to a parent fieldset
+   * @param {HTMLNode} questionRadios An array of dom elements.
+   * @returns {boolean} The status of the group of inputs
+   */
+  function checkQuestionState( questionRadios ) {
+    let radioIsChecked = false;
+
+    for ( let i = 0, l = questionRadios.length; i < l; i++ ) {
+      if ( questionRadios[i].checked ) {
+        radioIsChecked = true;
+      }
+    }
+
+    return radioIsChecked;
+  }
+
+  /**
+   * Checks the status of each fieldset to determine if
+   * the user has completed the form
+   * @returns {boolean} The completion status of the form
+   */
+  function checkformCompletion() {
+    let formCompleted = true;
+
+    Object.keys( questionStates ).forEach( function( key ) {
+      if ( !questionStates[key] ) {
+        formCompleted = false;
+      }
+    } );
+
+    return formCompleted;
+  }
+
+  /**
+   * Event handler to prevent clicking the submit button before
+   * the form is completed
+   * @param {MouseEvent} event The event object for the mousedown event.
+   */
+  function handleDisabledSubmit( event ) {
+    event.preventDefault();
+  }
```

Comment:
Is it really necessary to break this out into its own function? Is it essential for testing it the right way or something?

Reply:
Yes because we stop listening for it below.